src/components/TopScrollingBar.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+
+interface DoubleScrollbarProps {
+  children: React.ReactNode;
+}
+const TopScrollingBar: React.FC<DoubleScrollbarProps> = ({ children }) => {
+  const [width, setWidth] = useState<string>('auto');
+  const outerDivRef = useRef<HTMLDivElement | null>(null);
+  const childrenWrapperRef = useRef<HTMLDivElement | null>(null);
+
+  const getChildWrapperWidth = useCallback(() => {
+    const childrenWrapper = childrenWrapperRef.current;
+    const table = childrenWrapper?.getElementsByTagName('table')[0];
+    return table ? table.scrollWidth + 'px' : null;
+  }, []);
+
+  const setInnerDivWidth = useCallback(() => {
+    const newWidth = getChildWrapperWidth();
+    if (newWidth !== null && newWidth !== width) {
+      setWidth(newWidth);
+    }
+  }, [getChildWrapperWidth, width]);
+
+  useEffect(() => {
+    setInnerDivWidth();
+
+    const topScrollBarDiv = outerDivRef.current;
+    const tableWrapper: HTMLDivElement | null | undefined =
+      childrenWrapperRef.current?.querySelector('[datatype="table-wrapper"]');
```

Comment:
Can we use 'ref' instead of this as it seems to be more efficent and are aware of react componet life cycle whereas  querySelector is an old style which scans all dom to get hold of elements.  It is not that, using querySelector is not correct or break but just a suggestion to use something in react world while we are using the framework.